hummingbird/ml/_container.py

```diff
@@ -40,139 +44,43 @@ def __init__(self, sparkml_model):
-                splits = [df.to_numpy().reshape(-1, 1) for df in splits]
-                inputs = tuple(splits)
-            inputs = [*inputs]
-            variable_map = {}
-            device = _get_device(self)
-
-            # Maps data inputs to the expected variables.
-            for i, input_name in enumerate(self._input_names):
-                if type(inputs[i]) is list:
-                    inputs[i] = np.array(inputs[i])
-                if type(inputs[i]) is np.ndarray:
-                    inputs[i] = torch.from_numpy(inputs[i])
-                    if inputs[i].dtype == torch.float64:
-                        # We convert double precision arrays into single precision. Sklearn does the same.
-                        inputs[i] = inputs[i].float()
-                elif type(inputs[i]) is not torch.Tensor:
-                    raise RuntimeError("Inputer tensor {} of not supported type {}".format(input_name, type(inputs[i])))
-                if device is not None and device.type != "cpu":
-                    inputs[i] = inputs[i].to(device)
-                variable_map[input_name] = inputs[i]
-
-            # Evaluate all the operators in the topology by properly wiring inputs \ outputs
-            for operator in self._operators:
-                pytorch_op = self._operator_map[operator.full_name]
-                pytorch_outputs = pytorch_op(*(variable_map[input] for input in operator.input_full_names))
-
-                if len(operator.output_full_names) == 1:
-                    variable_map[operator.output_full_names[0]] = pytorch_outputs
-                else:
-                    for i, output in enumerate(operator.output_full_names):
-                        variable_map[output] = pytorch_outputs[i]
-
-            # Prepare and return the output.
-            if len(self._output_names) == 1:
-                return variable_map[self._output_names[0]]
-            else:
-                return list(variable_map[output_name] for output_name in self._output_names)
-
-
-class PyTorchTorchscriptSklearnContainer(ABC):
-    """
-    Base container for PyTorch and TorchScript models.
-    The container allows to mirror the Sklearn API.
-    """
+        Base container abstract class allowing to mirror the Sklearn API.
+        Thanks to *SklearnContainer* we can use the `predict`, `predict_proba` etc. API of Sklearn
+        also over the models generated by Hummingbird (irrespective of the selected backend).
 
-    def __init__(self, model, extra_config={}):
-        """
         Args:
-            model: A pytorch or torchscript model
-            extra_config: Some additional configuration parameter
+            model: A model
+            n_threads: How many threads should be used by the containter to run the model. None means use all threads.
+            batch_size: If different than None means split the input into batch_size partitions and score one partition at a time.
```

Comment:
the word "means" is confusing.
"If different than None, split the input into batc..."